public/app/plugins/datasource/cloudwatch/components/ConfigEditor.tsx

```diff
@@ -1,339 +1,63 @@
-import React, { PureComponent } from 'react';
-import { InlineFormLabel, LegacyForms, Button } from '@grafana/ui';
-const { Select, Input } = LegacyForms;
-import {
-  AppEvents,
-  SelectableValue,
-  DataSourcePluginOptionsEditorProps,
-  onUpdateDatasourceJsonDataOptionSelect,
-  onUpdateDatasourceResetOption,
-  onUpdateDatasourceJsonDataOption,
-  onUpdateDatasourceSecureJsonDataOption,
-} from '@grafana/data';
+import React, { FC, useEffect, useState } from 'react';
+import { Input, InlineField } from '@grafana/ui';
+import { DataSourcePluginOptionsEditorProps, onUpdateDatasourceJsonDataOption } from '@grafana/data';
+import { ConnectionConfig } from '@grafana/aws-sdk';
+
 import { getDatasourceSrv } from 'app/features/plugins/datasource_srv';
-import { CloudWatchDatasource } from '../datasource';
-import { CloudWatchJsonData, CloudWatchSecureJsonData } from '../types';
-import { CancelablePromise, makePromiseCancelable } from 'app/core/utils/CancelablePromise';
-import { appEvents } from 'app/core/core';
+import { store } from 'app/store/store';
+import { notifyApp } from 'app/core/actions';
+import { createWarningNotification } from 'app/core/copy/appNotification';
 
-const authProviderOptions = [
-  { label: 'AWS SDK Default', value: 'default' },
-  { label: 'Access & secret key', value: 'keys' },
-  { label: 'Credentials file', value: 'credentials' },
-] as SelectableValue[];
+import { CloudWatchJsonData, CloudWatchSecureJsonData } from '../types';
+import { CloudWatchDatasource } from '../datasource';
 
 export type Props = DataSourcePluginOptionsEditorProps<CloudWatchJsonData, CloudWatchSecureJsonData>;
 
-export interface State {
-  regions: SelectableValue[];
-}
+export const ConfigEditor: FC<Props> = (props: Props) => {
+  const [datasource, setDatasource] = useState<CloudWatchDatasource>();
 
-export class ConfigEditor extends PureComponent<Props, State> {
-  constructor(props: Props) {
-    super(props);
+  const addWarning = (message: string) => {
+    store.dispatch(notifyApp(createWarningNotification('CloudWatch Authentication', message)));
+  };
 
-    this.state = {
-      regions: [],
-    };
-  }
+  useEffect(() => {
+    getDatasourceSrv()
+      .loadDatasource(props.options.name)
```

Comment:
This should not return anything when you are first adding a datasource (cause there just isn't any saved yet).

Reply:
Yes! That's why there's the standard regions. 

So initially I added props.options as a dependency to the useEffect function. But then the regions are reloaded from the backend even if the datasource test wasn't successful. That's kind of annoying. Ideally, we'd had something on the props that would indicate if the test datasource function was successfull or not, but we don't have that today unfortunately.